src/main/java/Duke.java

```diff
@@ -1,10 +1,325 @@
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+
 public class Duke {
+
+    // shows if the Duke chat bot has been activated
+    private boolean activated;
+    // Line Separator
+    private final String SEP_LINE = "____________________________________________________________";
+    // Standard boot response
+    private final String bootMessage =
+            SEP_LINE
+            .concat(
+            " \n____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n"
+            + "\nMade by Dr-Octavius\n")
+            .concat(SEP_LINE)
+            .concat(
+            "\nHello! I'm Duke\n"
+            + "What can I do for you?\n")
+            .concat(SEP_LINE);
+    // Level-6 -> A-Collections: Text Storage
+    private final List<Task> storage = new ArrayList<>();
+
+    // Duke Constructor
+    public Duke() {
+        this.activated = true;
+    }
+
+    // Checks if Duke chat bot has been activated
+    public boolean isActive() {
+        return this.activated;
+    }
+
+    // Level-1: Greets user when Duke is initialised
+    public void greet() {
+        System.out.println(this.bootMessage);
+    }
+
+    // Level-1: echos user message back to user
+    // Level-5: returns the error message for unknown inputs
+    public String echo() {
+        return "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    }
+
+    // Level-1: Exit Message triggered by "bye"
+    public String exit() {
+        this.activated = false;
+        return " Bye. Hope to see you again soon!";
+    }
+
+
+    // Level-2: adds new task to user
+    public String add(String text) {
+        Task nextTask = new Task(text);
+        this.storage.add(nextTask);
+        return "added: ".concat(text);
+    }
+    // Level-2: Lists all items in storage
+    public String list() {
+        String out = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.storage.size(); i++) {
+            if (i != 0) out = out.concat("\n");
+            out = out.concat((i+1) + ".").concat(this.storage.get(i).toString());
+        }
+        return out;
+    }
+
+    // Level-3: Mark items as done
+    public String done(int index) {
+        this.storage.get(index).setDone();
+        Task task = this.storage.get(index);
+        return "Nice! I've marked this task as done: \n  ".concat(task.toString());
+    }
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
```

Comment:
Similarly should this be called e.g. `addTodoTask`? I notice the same issue in the next 2 functions too.